engine/execution/storehouse/in_memory_register_store.go

```diff
@@ -0,0 +1,333 @@
+package storehouse
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/onflow/flow-go/engine/execution"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+var _ execution.InMemoryRegisterStore = (*InMemoryRegisterStore)(nil)
+
+var ErrNotExecuted = fmt.Errorf("block is not executed")
+
+type PrunedError struct {
+	PrunedHeight uint64
+	Height       uint64
```

Comment:
can you add a comment explaining what is the difference between height and pruned height